### ISSUE TITLE
Severe performance drop after updating

### ISSUE THREAD

**Issue by TheProgrammerIncarnate** [tensorflow/tensorflow#37956]
**System information** 
- Fairly basic Keras/python script
- OS: Manjaro, up to date (as of 3/26)
- Device: Running on CPU (Intel 8th gen)
- TensorFlow (opt-cuda) installed from PacMan
-TF version: 2.1.0-4
- Python version: 3.8.2-1

**Explanation of issue**
I updated my system earlier today. A few minutes ago I started training a small feed-forward network using a python/Keras script. I had run this script before on a 95% identical training dataset, each epoch took about five seconds according to logs I saved. This time they took about four times as long. No other settings were changed. I double checked with a non-modified dataset and the results were the same.

According to logs, TensorFlow was previously on version 2.1.0-2 (now 2.1.0-4) The performance delta isn't caused by having other programs competing for CPU/RAM, or frequency/thermal throttling. I suppose minor updates to the kernel could cause some performance changes, but I don't see these causing such a profound difference.

I'd normally not care about a small performance drop, but since this was quite sizable I decided it was worth reporting.

**Comment by Saduf2019** [comment]
@TheProgrammerIncarnate
Could you please share the code run by you along with complete error logs for us to analyse

**Comment by TheProgrammerIncarnate** [comment]
My script is about 350 LoC total, so I condensed it down into only the training-related sections. I'll get back to you shortly with any error logs

```
#Hyperparameter & constant definitions
self.trainingBatchSize = 256
self.inferenceBatchSize = 65536
self.viewRadius = 4
self.hiddenLayers = 8
self.networkFraction = 0.8
self.dropoutFraction = 0.0#Dropout caused problems with this particular dataset
self.valSplit = 0.3
self.trainingPasses = 50

#Create four arrays for training and testing the network
#trainingPixels is based on input image
d = self.viewRadius * 2 + 1
trainingInputs = numpy.zeros((int((self.trainingPixels * (1 - self.valSplit))) +50,d,d,3),dtype=numpy.uint8)
trainingAnswers = numpy.zeros(int(self.trainingPixels * (1 - self.valSplit)) + 50)
testingInputs = numpy.zeros((int((self.trainingPixels * self.valSplit)) + 50,d,d,3),dtype=numpy.uint8)
testingAnswers = numpy.zeros(int(self.trainingPixels * self.valSplit) + 50)

#==Loops that fill arrays with data here==

#Setup model & data augmentation then run training
self.model = Sequential()
self.model.add(Flatten(input_shape=(d, d, 3)))
self.model.add(GaussianNoise(0.5))
while(self.hiddenLayers>0):
	self.hiddenLayers = self.hiddenLayers - 1
	self.model.add(Dense(int(totalInputs*self.networkFraction), activation='relu'))
	self.model.add(Dropout(self.dropoutFraction))
self.model.add(Dense(len(self.targetColors)+1, activation='softmax'))
dg = ImageDataGenerator(horizontal_flip=True,vertical_flip=True,data_format="channels_last")
self.model.compile(loss='sparse_categorical_crossentropy', optimizer='adam', metrics=['accuracy'])
modelInfo = self.model.fit_generator(
			dg.flow(trainingInputs,trainingAnswers,self.trainingBatchSize),
			validation_data=(testingInputs,testingAnswers),
			steps_per_epoch=math.ceil(self.trainingPixels / self.trainingBatchSize),
			epochs=trainingPasses,
			verbose=2)
```

**Comment by TheProgrammerIncarnate** [comment]
Here's the terminal output from running it (up to a few epochs)

`==Top of log==
Using TensorFlow backend.
2020-03-27 13:08:53.606732: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE3 SSE4.1 SSE4.2 AVX AVX2 FMA
2020-03-27 13:08:53.612068: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2208000000 Hz
2020-03-27 13:08:53.613027: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x562af8ccec80 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2020-03-27 13:08:53.613041: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2020-03-27 13:08:53.613151: I tensorflow/core/common_runtime/process_util.cc:147] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.
Epoch 1/50
 - 23s - loss: 0.4055 - accuracy: 0.8453 - val_loss: 0.2400 - val_accuracy: 0.9239
Epoch 2/50
 - 21s - loss: 0.2194 - accuracy: 0.9296 - val_loss: 0.1794 - val_accuracy: 0.9424
Epoch 3/50

`
Some other notes, the message about supported vector instructions has always shown up (I have not compiled from source to target this CPU.) The message about creating a thread pool is new however. Note: my task monitor shows all threads being fully utilized when training, same as before.

**Comment by Saduf2019** [comment]
@TheProgrammerIncarnate 
i do not follow what is meant by versions 2.1.0-2 (now 2.1.0-4) , i have replicated the code shared by you, please find the [gist here](https://colab.sandbox.google.com/gist/Saduf2019/ebb83ff7ae4edc55d6f7792c91beb719/37956.ipynb)

please share complete details for us to replicate the issue. if possible please share a colab gist for us to analyse the issue faced.

**Comment by TheProgrammerIncarnate** [comment]
@Saduf2019 
After poking around a little more, I believe the issue is out of Tensorflow's scope. The drop in performance is apparently caused by updating the python-tensorflow package from release 2 to release 4 (source version is 2.1.0 either way.) Thus this is something I need to talk to the package maintainer about (perhaps something to do with default optimization settings?)

**Comment by Saduf2019** [comment]
@TheProgrammerIncarnate
In that case please confirm if we may move this issue to closed status

**Comment by TheProgrammerIncarnate** [comment]
Ok, closing.
